kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,159 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Implementation of MPCA->Feature Selection->Linear SVM/LogisticRegression Pipeline
+"""
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.feature_selection import f_classif
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn.utils.validation import check_is_fitted
+
+from ..embed.mpca import MPCA
+
+classifiers = {"svc": [SVC, {"kernel": ["linear"], "C": np.logspace(-3, 2, 6)}],
+               "lr": [LogisticRegression, {"C": np.logspace(-3, 2, 6)}]}
+
+default_search_params = {'cv': 5}
```

Comment:
We did not do cv in matlab, see https://uk.mathworks.com/help/stats/fitclinear.html#d123e314573
Is it an option here or a must? It may not be necessary, considering how Matlab deals with it (and we get better results). CV on small sample may overfit as well.

Reply:
CV is used to determine the value of `C` only if `classifier_param` is set to be "auto".

Reply:
OK. Some light documentation/comments in the `tests` will be helpful for review / reading / future changes.